rust/connlib/clients/android/lib/build.gradle.kts

```diff
@@ -85,6 +85,11 @@ cargo {
     module  = "../"
     libname = "connlib"
     targets = listOf("arm", "arm64", "x86", "x86_64")
+    // Cargo requires both `targetDirectory` and `CARGO_TARGET_DIR` env to be set to work properly.
+    targetDirectory = "${project.buildDir}/cargo-target"
+    exec = { spec, _ ->
+        spec.environment("CARGO_TARGET_DIR", targetDirectory)
+    }
```

Comment:
@jamilbk @francesca64 
Without the `targetDirectory`, it used to output the `.SO` files to `rust/target` dir, hence the plugin could not set the correct `symlink`, updating it to this internal build path, it works as expected. I think this was due to some internal bug with the plugin, and can revisit this later if required.

Reply:
Ah gotcha, I wonder if it's a limitation with Android traversing paths outside the projectDir. In any case I think this is fine for now unless @francesca64 objects.

Reply:
I definitely object; this is very surprising.

Overriding the target directory is very unusual, and because there's no official way for custom build scripts to get the target directory I'd expect some libraries and tools in the wild to break in unexpected ways.

Since this isn't workspace-wide, it would also result in two target directories being used, even for Android alone. Besides being generally confusing (i.e. `cargo clean` wouldn't clean the gradle-specific one) this needlessly increases build times and storage usage (target directories can get shockingly large, especially if artifacts from multiple Rust versions accumulate).

Changing the target directory is absolutely a last resort. In `cargo-mobile` I just symlinked the `.so`s into the Android project, though even just automatically copying them would be perfectly fine.